katsdptelstate/rdb_reader.py

```diff
@@ -49,23 +50,35 @@ def end_sorted_set(self, key):
         self._zset = {}
 
 
-def load_from_file(client, filename):
+def load_from_file(callback, file):
     """Load keys from the specified RDB-compatible dump file.
 
     Parameters
-    ---------
-    client : :class:`redis.StrictRedis`-like
-        Redis client wrapper
-    filename : str
-        Filename of .rdb file to import
+    ----------
+    callback : :class:`rdbtools.RdbCallback`-like
+        Backend-specific callback that stores keys as RDB file is parsed
+    file : str or file-like object
+        Filename of .rdb file to import, or object representing contents of RDB
 
     Returns
     -------
     int
         Number of keys loaded
     """
-    callback = _Callback(client)
+    try:
+        size = os.path.getsize(file)
+    except TypeError:
+        start_pos = file.tell()
+        file.seek(0, 2)
+        size = file.tell() - start_pos
+        file.seek(start_pos)
+    except AttributeError:
+        size = -1   # File object does not support seeking
```

Comment:
I don't see how this AttributeError will be generated if the file doesn't support seeking, since the seek isn't in a `try` block. You also need to handle [IOError](https://docs.python.org/2/library/io.html#io.IOBase.seekable) and [OSError](https://docs.python.org/3/library/io.html?highlight=seekable#io.IOBase.seekable).

Given that the log message is only at debug level anyway and so isn't normally going to be seen, is it really worth this hassle to try to find the size?

Reply:
Also you could just set `size = 'unknown'` here rather than using -1 and then mapping -1 to 'unknown' in the log message.

Reply:
Simplified this to work with filenames only.